card/lib/card/name.rb

```diff
@@ -41,5 +41,17 @@ def trait tag_code
     def code
       Card::Codename[ Card.fetch_id self ]
     end
+
+
+    def is_a_field_of? context_name
```

Comment:
Ok, I'm not totally sure I get the logic of this.  And, if it works as the name implies, I think we should push it to the general name layer (Smarname for now).  We'll still keep it in a gem, but combine that gem in this repo as I understood it.  Well, never mind about the location, the logic ...

If the to_absolute does any expanding of _self, or _left, etc, or the standard +X expansion, then it can't match the relative name.  So, the magic in the line 49 must be producing a +X form from any prefix of the context card.  In effect we are testing independence from the context card.

I'm wondering whether we need to do the absolute name, won't the relative_name always have a leading + when then don't match, and always not when they do?